test/performance/odp_l2fwd.c

```diff
@@ -791,6 +791,11 @@ static int create_pktio(const char *dev, int idx, int num_rx, int num_tx,
 
 	odp_pktio_config(pktio, &config);
 
+	/* Explicitly enable promisc mode for devices that support
+	 * since we need to perform L2 forwarding.
+	 */
+	odp_pktio_promisc_mode_set(pktio, true);
```

Comment:
`odp_pktio_capability_t.set_op.op.promisc_mode` can be used to check if promisc mode is supported. After checking the capability, `odp_pktio_promisc_mode_set()`return value should also be checked.

The remaining question is do we want to always enable promiscuous mode and hence change the application default behavior? To maintain current default behavior promiscuous mode could be enabled using a command line option.

Reply:
Ok. I can check for the promisc mode set op capability and return value. But do we need to leave it as an option ?
Ideally shouldn't we have L2 fwd with promic mode ?

example/l2fwd_simple also by default enables promisc

Reply:
We are using  l2fwd app widely for testing and would like to keep the processing similar than what a "normal" application would do. In most cases promisc mode wouldn't be enabled.

Also, making promisc mode default would decrease CI coverage since packets with invalid MAC addresses would be accepted and possible problems in the packet generator application would go unnoticed.

Reply:
Agree with Matias that promisc mode should not be required/used by default, but should be a command line option. For example, "-P" character could be used for it "-P, --promisc_mode".